kratos/utilities/rbf_shape_functions_utility.h

```diff
@@ -0,0 +1,136 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Sebastian Ares de Parga
+//
+
+#if !defined(KRATOS_rbf_shape_functions_utility_H_INCLUDED)
+#define  KRATOS_rbf_shape_functions_utility_H_INCLUDED
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/array_1d.h"
+#include "includes/define.h"
+#include "spaces/ublas_space.h"
+#include "utilities/dense_householder_qr_decomposition.h"
+
+namespace Kratos
+{
+
+/**
+ * @brief Radial Basis Functions utility to calculate shape function values
+ * This class uses Gaussian raidal basis functions to calculate
+ * the shape function values for a given value (i.e. norm of a point) with partition of unity.
+ */
+class KRATOS_API(KRATOS_CORE) RBFShapeFunctionsUtility
+{
+
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Dense space definition for the QR decomposition using in the solve
+    using DenseSpace = UblasSpace<double, Matrix, Vector>;
+
+    /// QR decomposition pointer definition
+    using DenseQRPointerType = typename DenseQRDecomposition<DenseSpace>::Pointer;
+
+    /// Kratos core QR decomposition type
+    using KratosCoreQRType = DenseHouseholderQRDecomposition<DenseSpace>;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Calculate the RBF value
+     * This function evaluates the Gaussian RBF for a norm
+     * @param x Norm of RBF argument (i.e. norm of radial vector)
+     * @param h Gaussian radius
+     * @return double The RBF value
+     */
+    static double EvaluateRBF(
+        const double x,
+        const double h);
```

Comment:
I just noted that we can make this `private`. I'm assuming that you don't need/want to keep it `public` though.